app/grandchallenge/archives/migrations/0017_archive_item_delete_permission_granting.py

```diff
@@ -0,0 +1,47 @@
+from django.contrib.auth.management import create_permissions
+from django.db import migrations
+
+
+def add_archive_item_delete_permissions(apps, schema_editor):
+    Archive = apps.get_model("archives", "Archive")  # noqa: N806
+    ArchiveItem = apps.get_model("archives", "ArchiveItem")  # noqa: N806
+    ArchiveItemGroupObjectPermission = apps.get_model(  # noqa: N806
+        "archives", "ArchiveItemGroupObjectPermission"
+    )
+    Permission = apps.get_model("auth", "Permission")  # noqa: N806
+
+    queryset = Archive.objects.order_by("-created").prefetch_related(
+        "editors_group", "items"
+    )
+    for app_config in apps.get_app_configs():
+        app_config.models_module = True
+        create_permissions(app_config, verbosity=0)
+    delete_permission = Permission.objects.get(
+        codename=f"delete_{ArchiveItem._meta.model_name}",
+    )
+
+    for archive in queryset.iterator():
+        ArchiveItemGroupObjectPermission.objects.bulk_create(
+            objs=[
+                ArchiveItemGroupObjectPermission(
+                    content_object=item,
+                    group=archive.editors_group,
+                    permission=delete_permission,
+                )
+                for item in archive.items.all()
+            ],
+            ignore_conflicts=True,

```

Comment:
If the DB is not empty and there already are items with the given permission (which can happen because new items get the delete permission assigned at creation), this will fail with an integrity error. Ignoring those conflicts does the trick, although it feels a bit strange to do.

Reply:
For my understanding - does that mean that only the conflicting rows are ignored? It doesn't ignore the entire batch?

Reply:
From the documentation `On databases that support it (all but Oracle), setting the ignore_conflicts parameter to True tells the database to ignore failure to insert any rows that fail constraints such as duplicate unique values.` 

To me his reads as only the conflicting rows will be ignored, but I will do some tests to verify

Reply:
Only the failing rows are ignored